credentialProvider/src/main/java/nl/nn/credentialprovider/CredentialFactory.java

```diff
@@ -0,0 +1,99 @@
+/*
+   Copyright 2021 Nationale-Nederlanden
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.credentialprovider;
+
+
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import nl.nn.credentialprovider.util.AppConstants;
+import nl.nn.credentialprovider.util.Misc;
+
+public class CredentialFactory {
+	protected Logger log = Logger.getLogger(this.getClass().getName());
+
+	private final String PROPERTY_CREDENTIAL_FACTORY="credentialFactory.class";
+	private final String DEFAULT_CREDENTIAL_FACTORY1=FileSystemCredentialFactory.class.getName();
+	private final String DEFAULT_CREDENTIAL_FACTORY2=WebSphereCredentialFactory.class.getName();
+
+	private ICredentialFactory delegate;
+
+	private static CredentialFactory self;
+
+	public static CredentialFactory getInstance() {
+		if (self==null) {
+			self=new CredentialFactory();
+		}
+		return self;
+	}
+
+	private CredentialFactory() {
+		String factoryClassName = AppConstants.getInstance().getProperty(PROPERTY_CREDENTIAL_FACTORY);
+		if (Misc.isNotEmpty(factoryClassName) && tryFactory(factoryClassName)) {
```

Comment:
Deze Misc.IsNotEmpty is dubbel en hoort thuis in de tryFactory methode. (waar hij dus ook al in zit)